File: FinanceApp.Shared/Dto/Spending/CreateSpending.cs
using FinanceApp.Shared.Dto.Base;
using FinanceApp.Shared.Enum;

namespace FinanceApp.Shared.Dto.Spending
{
    public class CreateSpending : CreateDto
    {
        public string Name { get; set; }
        public double Amount { get; set; }
        public DateTime InitialDate { get; set; }
        public EPayment Payment { get; set; }
        public int? CreditCardId { get; set; }
        public int Recurrence { get; set; }
        public bool IsRequired { get; set; }
        public int? Category { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsEndless { get; set; }
        public int? TimesRecurrence { get; set; }

    }
}

File: FlutterApp/finance_app/lib/models/spending/create_spending.dart
class CreateSpending {
  String name;
  int recurrence;
  double amount;
  DateTime initialDate;
  DateTime? endDate;
  int? creditCardId;
  int? categoryId;
  bool isRequired;
  bool isEndless;
  int timesRecurrence;
  int payment;

  CreateSpending(
      {required this.name,
      required this.recurrence,
      required this.amount,
      required this.initialDate,
      required this.endDate,
      required this.payment,
      required this.creditCardId,
      required this.categoryId,
      required this.isRequired,
      required this.isEndless,
      required this.timesRecurrence});

  Map<String, dynamic> toJson() => {
        'name': name,
        'payment': payment,
        'recurrence': recurrence,
        'amount': amount,
        'initialDate': initialDate.toString(),
        'categoryId': categoryId,
        'isRequired': isRequired,
        'creditCardId': creditCardId,
        'endDate': endDate?.toString(),
        'isEndless': isEndless,
        'timesRecurrence': timesRecurrence,
      };
}
